src/ProtoBuf.FSharp/CodeGen.fs

```diff
@@ -19,6 +50,27 @@ let inline emitZeroValueOntoEvaluationStack (gen: ILGenerator) (getterType: Meth
         gen.Emit(OpCodes.Ldc_I4_0) // Push length onto the stack.
         gen.Emit(OpCodes.Newarr, elementType) // Initialise array with length.
 
+let private emitStackTopZeroCheck (gen : ILGenerator) (topType : Type) =
```

Comment:
Can we have a summary description as to what this method does, and maybe some commentary on how it does it? Some questions that a reader may have.
- What is the equivalent code in F#/C# for this method?
- Why should this be used?